src/components/augmentedReality/ARObjectListItem.js

```diff
@@ -0,0 +1,94 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Alert, StyleSheet } from 'react-native';
+import { ListItem } from 'react-native-elements';
+
+import { colors, consts, normalize, texts } from '../../config';
+import { deleteObject, downloadObject, DOWNLOAD_TYPE, progressSizeGenerator } from '../../helpers';
+import { ScreenName } from '../../types';
+import { RegularText } from '../Text';
+import { Touchable } from '../Touchable';
+
+import { IconForDownloadType } from './IconForDownloadType';
+
+export const ARObjectListItem = ({ data, index, item, navigation, setData, showOnDetailPage }) => {
+  const { DOWNLOAD_TYPE: itemDownloadType, progressSize, title, totalSize, locationInfo } = item;
+
+  const onPress = async () => {
+    if (showOnDetailPage) {
+      navigation.navigate(ScreenName.ArtworkDetail, { data, index });
+      return;
+    }
+
+    if (itemDownloadType === DOWNLOAD_TYPE.DOWNLOADABLE) {
+      await downloadObject({
+        index,
+        data,
+        setData
+      });
+    } else if (itemDownloadType === DOWNLOAD_TYPE.DOWNLOADED) {
+      Alert.alert(
+        texts.settingsTitles.arListLayouts.alertTitle,
+        texts.settingsTitles.arListLayouts.deleteAlertMessage,
+        [
+          {
+            text: texts.settingsTitles.arListLayouts.deleteAlertButtonText,
+            onPress: async () => {
+              await deleteObject({
+                index,
+                data,
+                setData
+              });
```

Comment:
we could one-line `await downloadObject({ index, data, setData });` and `await deleteObject({ index, data, setData });` for readability with less lines in whole file, right?

Reply:
https://github.com/ikuseiGmbH/smart-village-app-app/pull/563/commits/de0af1402bf2efa346b3afe2691ff4e2d35b2811